### DISCUSSION TITLE
Benchmarks for Source Generators

### DISCUSSION THREAD

**Discussion by papafe** [dotnet/roslyn#65254] [Q&A] [answered]
Is there any recommended approach for running benchmarks on Source Generators?

I would be interested in the running time of the source generation phase and see how it evolves with changes to the code.

**Comment by jasonmalinowski** [comment, marked as answer]
A couple of options: the compiler has a -reportanalyzer flag which will report time to the console; that will include generator time. You can do /p:ReportAnalyzer=true on the msbuild command line to pass that along.

If you want real-time reporting (say if you're typing in Visual Studio) there's also https://github.com/chsienki/GeneratorTracer which updates in real time based on ETW events we publish.

**Comment by papafe** [comment]
Thanks a lot for the suggestions!
